src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Wazuh Vulnerability scanner - Database Feed Manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * Oct 3, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _UPDATE_CVE_DESCRIPTION_HPP
+#define _UPDATE_CVE_DESCRIPTION_HPP
+
+#include "chainOfResponsability.hpp"
+#include "cve5_generated.h"
+#include "eventContext.hpp"
+#include "flatbuffers/flatbuffers.h"
+#include "rocksDBWrapper.hpp"
+#include "vulnerabilityDescription_generated.h"
+
+constexpr auto DESCRIPTION_DATABASE_PATH {"queue/vd/descriptions"};
+constexpr auto CVE5_DATABASE_PATH {"queue/vd/cve5"};
+
+/**
+ * @brief UpdateCVEDescription class.
+ *
+ */
+class UpdateCVEDescription final
+{
+public:
+    /**
+     * @brief Reads CVE5 database, creates a vulnerability description flatbuffer and stores it in a specific RocksDB
+     * database.
+     *
+     * @param resource CVE ID.
+     * @param cve5Flatbuffer CVE5 Flatbuffer.
+     */
+    static void storeVulnerabilityDescription(const std::string& resource, const cve_v5::Entry* cve5Flatbuffer)
+    {
+        auto metricsArray = cve5Flatbuffer->containers()->cna()->metrics();
+        const size_t sizeM = (nullptr != metricsArray) ? metricsArray->size() : 0;
+        auto descriptionArray = cve5Flatbuffer->containers()->cna()->descriptions();
+        const size_t sizeD = (nullptr != descriptionArray) ? descriptionArray->size() : 0;
+        auto referencesArray = cve5Flatbuffer->containers()->cna()->references();
+        const size_t sizeR = (nullptr != referencesArray) ? referencesArray->size() : 0;
+        size_t sizeT = (sizeM > sizeD) ? sizeM : sizeD;
+        sizeT = (sizeT > sizeR) ? sizeT : sizeR;
+
+        float VulnDescFBScoreBase = 0.0;
+        std::string VulnDescFBClassificationStr;
+        std::string VulnDescFBDescriptionStr;
+        std::string VulnDescFBSeverityStr;
+        std::string VulnDescFBScoreVersionStr;
+        std::string VulnDescFBReferenceStr;
+
+        bool cont = true;
+        for (size_t i = 0; i < sizeT; ++i)
+        {
+            if (i < sizeM)
+            {
+                auto metric = metricsArray->Get(i);
+                if (metric->cvssV3_1())
```

Comment:
In case we don't find `cvssV3_1` , we should try `cvssV3_0` or lower?

Reply:
We can ignore this for now.

Reply:
It was a good catch, I thought about this but forgot to discuss it with the team. Anyway I added a logic for this case.